README.md

```diff
@@ -1 +1,96 @@
-# Reloader
\ No newline at end of file
+# RELOADER
+
+## WHY NAME RELOADER
+
+In english language, Reloader is a thing/tool that can reload certain stuff. So refereig to that meaning relaoder can reload
+
+## Problem
+
+We would like to watch if some change happens in `ConfigMap` and `Secret` objects and then perform certain upgrade on relavent `Deployment`, `Deamonset` and `Statefulset`
+
+## Solution
+
+Reloader can watch any changes in `ConfigMap` and `Secret` objects and then performs rolling upgrades on their associated `Deployments`, `Deamonsets` and `Statefulsets` and updating these dynamically.  
+
+**NOTE:** This controller has been inspired from [configmapController](https://github.com/fabric8io/configmapcontroller)
+
+## How to use Reloader
+
+For a `Deployment` called `foo` have a `ConfigMap` called `foo`. Then add this annotation to your `Deployment`
+
+```yaml
+metadata:
+  annotations:
+    reloader.stakater.com/update-on-change: "foo"
+```
+
+Then, providing `Reloader` is running, whenever you edit the `ConfigMap` called `foo` the Reloader will update the `Deployment` by adding the environment variable:
+
+```
+STAKATER_FOO_REVISION=${reloaderRevision}
+```
+
+This then triggers a rolling upgrade of your deployment's pods to use the new configuration.
+
+Same procedure can be followed to perform rolling upgrade on `Deamonsets` and `Statefulsets` as well.
```

Comment:
Daemonsets and Statefulsets don't preform rolling upgrades, do they? @faizanahmad055 @waseem-h 
Please verify and correct me if I'm wrong

Reply:
They do, but only if you have explicitly set its rolling update strategy to RollingUpdate